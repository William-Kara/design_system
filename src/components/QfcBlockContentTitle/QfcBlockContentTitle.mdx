import { ArgTypes, Meta } from '@storybook/blocks';
import QfcBlockContentTitle from '@/components/QfcBlockContentTitle/QfcBlockContentTitle.vue';

<Meta title="Prototypes/BlockContentTitle" />

# Definition

QfcBlockImgTitle prototype is used to display:

- title
- paragraph
- video
- CTA button
- tags (optional)

## Props

The following props are available for the QfcBlockContentTitle component:

<ArgTypes of={QfcBlockContentTitle} />

# Usage

To use QfcBlockImgTitle, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcBlockContentTitle.scss';
```

The button trigger a click event under the name of "button-click"

```html
<template>
  <QfcBlockImgTitle @button-click="$router.push('/unknown')" />
</template>
```

BlockContentTitle can also be used with template slots, allowing you to customize the title and paragraph with html like so

```html
<template>
  <QfcBlockImgTitle >
        <template #title>
            <b>Lorem</b> ipsum
        </template>
        <template #paragraph>
            Lorem <b>ipsum</b>
        </template>
      </QfcBlockContentTitle>
  <QfcBlockImgTitle />
</template>
```
