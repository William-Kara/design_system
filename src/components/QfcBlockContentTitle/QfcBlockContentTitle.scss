.qfc-block {
  max-width: 87.94rem;
  display: grid;
  grid-template-areas:
    'text content'
    'interactive content';
  grid-template-columns: 60% auto;
  grid-template-rows: auto 1fr;

  .qfc-block__text {
    padding-top: $q-generic-space-7 * 2;
    grid-area: text;

    &__title {
      font-size: $fs-xxl;
      font-weight: $semibold;
      line-height: 82px;
      margin-bottom: $q-generic-space-5;
    }
    &__paragraph {
      font-size: $fs-m;
      margin-bottom: $q-generic-space-5;
      width: 70%;
    }
  }

  .qfc-block__content {
    grid-area: interactive;
    &__tags {
      margin-bottom: $q-generic-space-8;
      overflow: visible;
    }
    &__button {
      width: fit-content;
    }
  }

  .qfc-block__media {
    grid-area: content;
    max-height: 730px;
    margin: auto;
  }
}

.qfc-block {
  @include tablet-only {
    grid-column-gap: $q-generic-space-7;
    grid-template-areas:
      'text content'
      'interactive';

    .qfc-block__text {
      padding-top: $q-generic-space-10;
      &__title {
        font-size: $fs-xl;
        line-height: 55px;
      }
      &__paragraph {
        width: 100%;
      }
    }

    .qfc-block__media {
      height: 100%;
      width: 100%;
    }
  }
}

.qfc-block {
  @include mobile {
    grid-template-areas:
      'content'
      'text'
      'interactive';
    grid-template-columns: 1fr;

    .qfc-block__text {
      padding-top: 0;
      margin-bottom: $q-generic-space-5;
      &__title {
        font-size: $fs-xl;
        line-height: 55px;
        margin-bottom: $q-generic-space-3;
      }
      &__paragraph {
        width: 100%;
        margin-bottom: 0;
      }
    }

    .qfc-block__content {
      &__tags {
        margin-bottom: $q-generic-space-3;
      }
      &__button {
        width: 100%;
        box-sizing: border-box;
      }
    }

    .qfc-block__media {
      justify-self: center;
      width: 64%;
    }
  }
}
