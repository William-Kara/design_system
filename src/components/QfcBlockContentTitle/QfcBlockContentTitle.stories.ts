import BlockImgTitle from './QfcBlockContentTitle.vue';
import type { Meta, StoryObj } from '@storybook/vue3';
import videoFile from '../../public/accueil-animation.mp4';

const meta: Meta<typeof BlockImgTitle> = {
  component: BlockImgTitle,
  title: 'Prototypes/BlockContentTitle',
  parameters: {},
  argTypes: {
    title: {
      type: 'string',
    },
    paragraph: {
      type: 'string',
    },
    buttonLabel: {
      type: 'string',
    },
    src: {
      type: 'string',
    },
    alt: {
      type: 'string',
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

const video = {
  src: videoFile,
};

export const Primary: Story = {
  args: {
    title: 'Nous construisons votre nouvelle plateforme de e-learning.',
    paragraph:
      'Vous souhaitez acquérir de nouvelles compétences ou simplement vous perfectionner sur un sujet en particulier ?',
    tagList: ['Cybersécurité', 'Développement', 'Design'],
    buttonLabel: 'Découvrez notre pédagogie',
    src: video.src,
    alt: 'video presentation',
  },
};
