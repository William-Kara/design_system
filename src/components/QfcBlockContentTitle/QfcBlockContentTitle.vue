<template>
  <div class="qfc-block">
    <div class="qfc-block__text">
      <h2 class="qfc-block__text__title"><slot name="title" />{{ title }}</h2>
      <p class="qfc-block__text__paragraph"><slot name="paragraph" />{{ paragraph }}</p>
    </div>

    <div class="qfc-block__content">
      <QfcTagList v-if="tagList.length" class="qfc-block__content__tags" :tag-list="tagList" />
      <QfcButton
        class="qfc-block__content__button"
        :label="buttonLabel"
        :icon="false"
        size="medium"
        @click="$emit('button-click')"
      />
    </div>

    <video class="qfc-block__media" :src="src" loop autoplay>
      Sorry, your browser doesn't support HTML5 video tag.
    </video>
  </div>
</template>

<script lang="ts" setup>
import QfcTagList from '@/components/QfcTagList/QfcTagList.vue';
import QfcButton from '@/components/QfcButton/QfcButton.vue';

type Props = {
  title: string;
  paragraph: string;
  tagList: string[];
  buttonLabel: string;
  src: string;
  alt: string;
};

defineProps<Props>();
</script>

<style lang="scss" scoped src="./QfcBlockContentTitle.scss" />
