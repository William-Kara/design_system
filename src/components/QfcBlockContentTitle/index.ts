import QfcBlockContentTitle from './QfcBlockContentTitle.vue';

export { QfcBlockContentTitle };
export default QfcBlockContentTitle;
