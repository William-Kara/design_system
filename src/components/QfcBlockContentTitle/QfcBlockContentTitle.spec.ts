import { it, describe, beforeEach, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import QfcBlockContentTitle from './QfcBlockContentTitle.vue';
import videoFile from '@/public/accueil-animation.mp4';

let wrapper: any = null;

const propsValues = {
  title: 'Nous construisons votre nouvelle plateforme de e-learning.',
  paragraph:
    'Vous souhaitez acquérir de nouvelles compétences ou simplement vous perfectionner sur un sujet en particulier ?',
  tagList: ['Cybersécurité', 'Développement', 'Design'],
  buttonLabel: 'Découvrez notre pédagogie',
  src: videoFile,
  alt: 'video presentation',
};

beforeEach(() => {
  wrapper = shallowMount(QfcBlockContentTitle, {
    propsData: propsValues,
  });
});

describe('QfcBlockContentTitle Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
