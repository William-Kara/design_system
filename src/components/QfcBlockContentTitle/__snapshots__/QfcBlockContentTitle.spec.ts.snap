// Vitest Snapshot v1

exports[`QfcBlockContentTitle Component > should render correctly 1`] = `
"<div data-v-890b97bd=\\"\\" class=\\"qfc-block\\">
  <div data-v-890b97bd=\\"\\" class=\\"qfc-block__text\\">
    <h2 data-v-890b97bd=\\"\\" class=\\"qfc-block__text__title\\">Nous construisons votre nouvelle plateforme de e-learning.</h2>
    <p data-v-890b97bd=\\"\\" class=\\"qfc-block__text__paragraph\\">Vous souhaitez acquérir de nouvelles compétences ou simplement vous perfectionner sur un sujet en particulier ?</p>
  </div>
  <div data-v-890b97bd=\\"\\" class=\\"qfc-block__content\\">
    <qfc-tag-list-stub data-v-890b97bd=\\"\\" taglist=\\"Cybersécurité,Développement,Design\\" class=\\"qfc-block__content__tags\\"></qfc-tag-list-stub>
    <qfc-button-stub data-v-890b97bd=\\"\\" icon=\\"false\\" isdisabled=\\"false\\" label=\\"Découvrez notre pédagogie\\" secondary=\\"false\\" size=\\"medium\\" class=\\"qfc-block__content__button\\"></qfc-button-stub>
  </div><video data-v-890b97bd=\\"\\" class=\\"qfc-block__media\\" src=\\"/public/accueil-animation.mp4\\" loop=\\"\\" autoplay=\\"\\"> Sorry, your browser doesn't support HTML5 video tag. </video>
</div>"
`;
